File: TP_01/Q08/Soma.java


import java.util.Scanner;

public class Soma {
    
    
    public static int SomaTermos(String num){

        return SomaTermos(num, 0, 0);
    }

    public static int SomaTermos(String num, int i, int sum){
        
        if(i == num.length()){

            return sum;
        } else {

            char c = num.charAt(i);
            if(c >= '0' && c <= '9'){

                sum += (int) c - '0';
            }
            
            return SomaTermos(num, i + 1, sum);
        }
        
    }

    public static void main(String[] args) {
        
        // Definir dados
        Scanner scanner = new Scanner(System.in);
        String n;

        // Leitura de dados
        n = scanner.nextLine();

        while(!n.equals("FIM")){

            // Mostrar dados
            System.out.println(SomaTermos(n));

            n = scanner.nextLine();
        }
        
        

        scanner.close();
    }
}

File: TP_01/Q08/Soma.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include<string.h>
#include<ctype.h>


int SomaTermos(char* num, int i, int sum){

    // Definir dados
    int len = strlen(num);

    if(i == len){

        return sum;
    } else {

        sum += (int) num[i] - '0';

        return SomaTermos(num, i + 1, sum);
    }
}

int main(){

    // Definir dados
    int num;
    char str[256];

    fgets(str, 256, stdin);
    str[strcspn(str, "\n")] = '\0';

    while(strcmp(str, "FIM") != 0){

        printf("%d\n", SomaTermos(str, 0, 0));

        fgets(str, 256, stdin);
        str[strcspn(str, "\n")] = '\0';
    }
    

    return 0;
}